Updated details on maintenance updates schedule

runwuf opened the pull request:
Update documentation to explain the schedule of optional and mandatory updates.
slack thread [here](https://aiven-io.slack.com/archives/C02BW1F2WG7/p1675184919227259).

staceysalamon-aiven commented on docs/platform/concepts/maintenance-window.rst:
```diff
@@ -17,3 +17,11 @@ In case of **Apache Kafka®** and **OpenSearch®** the service DNS address resol
+
+
+Maintenance updates
+===================
```
```suggestion
~~~~~~~~~~~~~~~~~~~
```

staceysalamon-aiven replied:
Changing this to create a heading instead of page title.

staceysalamon-aiven commented on docs/platform/concepts/maintenance-window.rst:
```diff
@@ -17,3 +17,11 @@ In case of **Apache Kafka®** and **OpenSearch®** the service DNS address resol
+Maintenance updates
+===================
+
+Updates are initially available as optional, once it reaches the 6 months mark it becomes mandatory.
```
```suggestion
Updates are initially optional. After they have been available for six months, they become mandatory. Mandatory updates are applied with at least seven days advance notice with the exception of critical security updates, which are applied in the maintenance window of the current week.
```

staceysalamon-aiven replied:
Reworded for clarity.

staceysalamon-aiven commented on docs/platform/concepts/maintenance-window.rst:
```diff
@@ -17,3 +17,11 @@ In case of **Apache Kafka®** and **OpenSearch®** the service DNS address resol
+
+Updates are initially available as optional, once it reaches the 6 months mark it becomes mandatory.
+
+Once updates are set to be mandatory, these updates would be applied on the next week's maintenance window or after, which gives at least 7 days in advance with exception of critical security related updates would be applied in the maintenance of the current week.
```
```suggestion
```

staceysalamon-aiven reviewed (changes requested):
Thanks for adding this @runwuf! I have just fixed the RST for the heading and made a suggestion to reword the language in the maintenance updates section. I will leave it then with the others in the original discussion to check this for accuracy.

runwuf commented:
Thanks for reviewing this PR. I added this based on the scenario @fingon described and updated the rest of the details as per @staceysalamon-aiven 's suggestion.

> Service quarterly patch updates are always mandatory with 6 months advance notice.

fingon commented on docs/platform/concepts/maintenance-window.rst:
```diff
@@ -17,3 +17,11 @@ In case of **Apache Kafka®** and **OpenSearch®** the service DNS address resol
+Maintenance updates
+~~~~~~~~~~~~~~~~~~~
+
+Service quarterly patch updates are always mandatory with 6 months advance notice.  Other updates are initially optional. 
```
```suggestion
Security updates, platform updates which affect reliability or stability of the service nodes, as well as quarterly patch updates are always mandatory with advance notice of week(s).  Other updates are initially optional. 
```

fingon replied:
(I wouldn't give exact numbers, but for PG quarterly the number is in 2-3 weeks range usually)

fingon commented:
Content LGTM now but I'm sure @staceysalamon-aiven can come up with better way to say it :)

staceysalamon-aiven commented on docs/platform/concepts/maintenance-window.rst:
```diff
@@ -17,3 +17,11 @@ In case of **Apache Kafka®** and **OpenSearch®** the service DNS address resol
+
+Maintenance updates
+~~~~~~~~~~~~~~~~~~~
+
```
Suggested rewording: 

Security updates, platform updates that affect reliability or stability of the service nodes, and quarterly patch updates are always mandatory. Other updates are initially optional. Advance notice is given for all updates. After optional updates have been available for six months, they become mandatory and are applied on the next week's maintenance window at the earliest. This means you have at least 7 days advance notice with exception of critical security updates. These critical updates are applied in the maintenance window of the current week.

runwuf replied:
Updated, thank you so much!

staceysalamon-aiven reviewed (changes requested):
I just made a suggestion to reword this a little for clarity. Thanks!
